src/machine.rs

```diff
@@ -626,9 +637,20 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
         } else {
             None
         };
+        let buffer_alloc = if ecx.machine.weak_memory {
+            // FIXME: if this is an atomic obejct, we want to supply its initial value
+            // while allocating the store buffer here.
```

Comment:
```suggestion
```
This is covered by the FIXMEs elsewhere (and this file should not contain relevant logic anyway).